store/precomputed_key/s3/s3.go

```diff
@@ -97,7 +99,16 @@ func (s *Store) Get(ctx context.Context, key []byte) ([]byte, error) {
 }
 
 func (s *Store) Put(ctx context.Context, key []byte, value []byte) error {
-	_, err := s.client.PutObject(ctx, s.cfg.Bucket, path.Join(s.cfg.Path, hex.EncodeToString(key)), bytes.NewReader(value), int64(len(value)), minio.PutObjectOptions{})
+	endpointURL, err := url.Parse(s.cfg.Endpoint)
+	if err != nil {
+		return err
+	}
+
+	putObjectOptions := minio.PutObjectOptions{}
+	if s3utils.IsGoogleEndpoint(*endpointURL) {
+		putObjectOptions.DisableContentSha256 = true // Avoid chunk signatures on GCS: https://github.com/minio/minio-go/issues/1922
+	}
```

Comment:
its more efficient to do this check within the constructor and embed `putObjectOptions` as a field into the storage struct versus as an additional logic flow that adds latency to every insertion